handlers/dimensions.go

```diff
@@ -1,27 +1,114 @@
 package handlers
 
 import (
+	"fmt"
 	"net/http"
+	"net/url"
+	"strings"
 
+	"github.com/ONSdigital/dp-api-clients-go/v2/filter"
 	"github.com/ONSdigital/dp-frontend-filter-flex-dataset/mapper"
 	"github.com/ONSdigital/dp-net/v2/handlers"
+	"github.com/ONSdigital/log.go/v2/log"
 	"github.com/gorilla/mux"
 )
 
 // DimensionsSelector Handler
-func DimensionsSelector(rc RenderClient) http.HandlerFunc {
+func DimensionsSelector(rc RenderClient, fc FilterClient, dc DimensionClient) http.HandlerFunc {
 	return handlers.ControllerHandler(func(w http.ResponseWriter, req *http.Request, lang, collectionID, accessToken string) {
-		dimensionsSelector(w, req, rc, lang)
+		dimensionsSelector(w, req, rc, fc, dc, collectionID, accessToken, lang)
 	})
 }
 
-func dimensionsSelector(w http.ResponseWriter, req *http.Request, rc RenderClient, lang string) {
+func dimensionsSelector(w http.ResponseWriter, req *http.Request, rc RenderClient, fc FilterClient, dc DimensionClient, collectionID, accessToken, lang string) {
+	ctx := req.Context()
 	vars := mux.Vars(req)
+	filterID := vars["filterID"]
+	nameParam := vars["name"]
+
+	logData := log.Data{
+		"filter_id":  filterID,
+		"name_param": nameParam,
+	}
+
+	currentFilter, _, err := fc.GetJobState(ctx, accessToken, "", "", collectionID, filterID)
+	if err != nil {
+		log.Error(ctx, "failed to get job state", err, logData)
+		setStatusCode(req, w, err)
+		return
+	}
+
+	logData["population_type"] = currentFilter.PopulationType
+
+	filterDimension, err := findDimension(nameParam, currentFilter.Dimensions)
+	if err != nil {
+		log.Error(ctx, "failed to find dimension in filter", err, logData)
+		setStatusCode(req, w, err)
+		return
+	}
+
+	logData["filter_name"] = filterDimension.Name
 
-	// TODO: Get name from endpoint
-	dimension := vars["name"]
 	basePage := rc.NewBasePageModel()
-	m := mapper.CreateSelector(req, basePage, dimension, lang)
 
-	rc.BuildPage(w, m, "selector")
+	if !isAreaType(filterDimension) {
+		selector := mapper.CreateSelector(req, basePage, filterDimension.Name, lang)
+		rc.BuildPage(w, selector, "selector")
+		return
+	}
+
+	areaTypes, err := dc.GetAreaTypes(ctx, accessToken, "", currentFilter.PopulationType)
+	if err != nil {
+		log.Error(ctx, "failed to get geography dimensions", err, logData)
+		setStatusCode(req, w, err)
+		return
+	}
+
+	selector := mapper.CreateAreaTypeSelector(req, basePage, lang, areaTypes.AreaTypes, nameParam)
+	rc.BuildPage(w, selector, "selector")
+}
+
+// isAreaType determines if the current dimension is an area type
+func isAreaType(dimension *filter.ModelDimension) bool {
+	if dimension == nil {
+		return false
+	}
+
+	if dimension.IsAreaType == nil {
+		return false
+	}
+
+	return *dimension.IsAreaType
```

Comment:
I dislike this... It is a design smell.
But this is legacy logic.

Reply:
I agree and I think we can avoid this being used in the handler as its used to determine the view. It might be easier/clean to have a `bool` on the view model that renders the different views